### DISCUSSION TITLE
react-hook-form + material UI Autocomplete weird behaviour

### DISCUSSION THREAD

**Discussion by DiogoBatista** [react-hook-form/react-hook-form#3590] [Q&A] [answered]
Hello! 

Im migrating to v6 and I just noticed a weird behaviour on something that was previously working just fine. 

Basically the material ui Autocomplete started to trow this error when I upgraded to v6.

```
Material-UI: The `getOptionLabel` method of Autocomplete returned undefined instead of a string for 0.
```

I've made a simple code sandbox (https://codesandbox.io/s/polished-water-rm0h6?) to replicate the issue, If you downgrade the react-hook-form for a version before v6, like v5.5 or so it will work just fine. 

Im wondering if I'm missing something since it's clearly a major update so probably some core functionality changed as well. 

Has anyone seen this behaviour before, some help would be awesome! 

Thank you and keep up the good work react-hook-form team I love this library :D use it every-time I need a form

**Comment by bluebill1049** [comment, marked as answer]
please use `render` instead 👍

**Reply by DiogoBatista** [reply]
@bluebill1049 Thank you for your quick reply but I've been trying to make sense of this and I don't seem to be able to find a way to make it work with the render. Maybe a bit of context from my side would be beneficial. 

I need to pass a default value to a multiple Autocomplete and only then let it be modified with new values. In the code sandbox (https://codesandbox.io/s/polished-water-rm0h6) you can see that I edited the code to use render instead but every time you press the submit button nothing get's passed to the form values.
And if we pass a default value then every time we submit we only have that defaultValue returned from the controlled component and not an array with two values as it would be expected.

Thank you very much for your help, I just want to understand if I'm using RHF wrongly or this might be something that is wrong with the Autocomplete component from material ui

Thank you

**Reply by DiogoBatista** [reply]
@bluebill1049 I think I might have found a previous code sandbox that shows how to use it properly. 
https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r?file=/src/MuiAutoComplete.js:253-883 I tested it and I can do all my use cases so I will try again and mark this as completed if it solved my problem! 

Once again thank you for all your help and awesome work 🙏👏
